src/packageurl/contrib/django_models.py

```diff
@@ -29,100 +29,12 @@
-from django.utils.translation import gettext_lazy as _
-
-from packageurl import PackageURL
-
-
-class PackageURLMixin(models.Model):
-    """
-    Abstract Model for Package URL "purl" fields support.
-    """
-    type = models.CharField(
-        max_length=16,
-        blank=True,
-        help_text=_(
-            'A short code to identify the type of this package. '
-            'For example: gem for a Rubygem, docker for a container, '
-            'pypi for a Python Wheel or Egg, maven for a Maven Jar, '
-            'deb for a Debian package, etc.'
-        )
-    )
-
-    namespace = models.CharField(
-        max_length=255,
-        blank=True,
-        help_text=_(
-            'Package name prefix, such as Maven groupid, Docker image owner, '
-            'GitHub user or organization, etc.'
-        ),
-    )
-
-    name = models.CharField(
-        max_length=100,
-        blank=True,
-        help_text=_('Name of the package.'),
-    )
-
-    version = models.CharField(
-        max_length=100,
-        blank=True,
-        help_text=_('Version of the package.'),
-    )
-
-    qualifiers = models.CharField(
-        max_length=1024,
-        blank=True,
-        help_text=_(
-            'Extra qualifying data for a package such as the name of an OS, '
-            'architecture, distro, etc.'
-        ),
-    )
-
-    subpath = models.CharField(
-        max_length=200,
-        blank=True,
-        help_text=_(
-            'Extra subpath within a package, relative to the package root.'
-        ),
-    )
-
-    class Meta:
-        abstract = True
-
-    @property
-    def package_url(self):
-        """
-        Return a compact Package URL "purl" string.
-        """
-        try:
-            purl = PackageURL(
-                self.type, self.namespace, self.name,
-                self.version, self.qualifiers, self.subpath
-            )
-        except ValueError:
-            return ''
-        return str(purl)
-
-    def set_package_url(self, package_url):
-        """
-        Set each field values to the values of the provided `package_url` string
-        or PackageURL object.
-        Existing values are always overwritten, forcing the new value or an
-        empty string on all the `package_url` fields since we do not want to
-        keep any previous values.
-        """
-        if not isinstance(package_url, PackageURL):
-            package_url = PackageURL.from_string(package_url)
-
-        package_url_dict = package_url.to_dict(encode=True, empty='')
-        for field_name, value in package_url_dict.items():
-            model_field = self._meta.get_field(field_name)
-
-            if value and len(value) > model_field.max_length:
-                message = _('Value too long for field "{}".'.format(field_name))
-                raise ValidationError(message)
-
-            setattr(self, field_name, value)
+import warnings
+
+# Backward compatibility
+from packageurl.contrib.django.models import *
```

Comment:
Any specific reason for the renaming?

Reply:
Yes, I've added a `filters.py` module and since it's Django related too, having a `django/` directory is much cleaner than `django_models.py`, `django_filters.py`, `django_*`.